From 7b1e54bb6c89b6735d54faefb067e193c109dfbb Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:02:53 -0400
Subject: [fs] btrfs: abort the transaction when we don't find our extent ref

Message-id: <1407179109-13714-287-git-send-email-sandeen@redhat.com>
Patchwork-id: 87250
O-Subject: [RHEL7.1 PATCH 286/422] Btrfs: abort the transaction when we don't find our extent ref
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Josef Bacik <jbacik@fb.com>

commit c4a050bbbb5d7dab03aa720af36d8e91ed7f2ec8
Author: Josef Bacik <jbacik@fb.com>
Date:   Fri Mar 14 16:36:53 2014 -0400

    Btrfs: abort the transaction when we don't find our extent ref

    I'm not sure why we weren't aborting here in the first place, it is obviously a
    bad time from the fact that we print the leaf and yell loudly about it.  Fix
    this up, otherwise we panic because our path could be pointing into oblivion.
    Thanks,

    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/extent-tree.c b/fs/btrfs/extent-tree.c
index 1341163..1306487 100644
--- a/fs/btrfs/extent-tree.c
+++ b/fs/btrfs/extent-tree.c
@@ -5745,6 +5745,8 @@ static int __btrfs_free_extent(struct btrfs_trans_handle *trans,
    "unable to find ref byte nr %llu parent %llu root %llu  owner %llu offset %llu",
    bytenr, parent, root_objectid, owner_objectid,
    owner_offset);
+  btrfs_abort_transaction(trans, extent_root, ret);
+  goto out;
  } else {
   btrfs_abort_transaction(trans, extent_root, ret);
   goto out;
-- 
1.7.1